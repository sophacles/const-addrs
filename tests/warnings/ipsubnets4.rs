use const_addrs::ipsubnets4;

fn main() {
    let _ = ipsubnets4!("10.0.1.0", "10.0.0.0", 24);

    // failure so output is compared
    let _ = ipsubnets4!("10.0.10", "10.0.0.0", 24);
}
