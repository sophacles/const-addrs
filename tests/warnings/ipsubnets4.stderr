warning: This range will yeild no values: IP1 >= IP2
 --> tests/warnings/ipsubnets4.rs:4:25
  |
4 |     let _ = ipsubnets4!("10.0.1.0", "10.0.0.0", 24);
  |                         ^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: This iterator requires the lower IP value first in it's arguments list.

error: invalid IP address syntax
 --> tests/warnings/ipsubnets4.rs:7:25
  |
7 |     let _ = ipsubnets4!("10.0.10", "10.0.0.0", 24);
  |                         ^^^^^^^^^
